spv/sync.go

```diff
@@ -318,7 +322,18 @@ func (s *Syncer) setRequiredHeight(tipHeight int32) {
 
 // Run synchronizes the wallet, returning when synchronization fails or the
 // context is cancelled.
-func (s *Syncer) Run(ctx context.Context) error {
+func (s *Syncer) Run(ctx context.Context) (err error) {
+	s.doneMu.Lock()
+	s.done = make(chan struct{})
+	s.err = nil
+	s.doneMu.Unlock()
+	go func() {
```

Comment:
Hmm, this doesn't look right.  It's unconditionally immediately launching a goroutine that closes the done channel and sets the err to `err` which will be nil.

This was probably intended to be a `defer`.

Reply:
yep